app/src/main/res/values/dimens.xml

```diff
@@ -540,7 +539,10 @@
 
   <dimen name="promoted_story_card_layout_margin_end">8dp</dimen>
   <dimen name="promoted_story_card_layout_margin_start">8dp</dimen>
-  <dimen name="promoted_story_card_layout_width">280dp</dimen>
+  <!-- The width is set to sp so that it will scale according to the device screen -->
+  <dimen name="promoted_story_card_layout_width">280sp</dimen>
+  <!-- The width is set to sp so that it will scale according to the device screen -->
+  <dimen name="promoted_story_card_width">280sp</dimen>
```

Comment:
Why aren't we using this solution elsewhere in the PR? How does a developer decide between using the minWidth/Height approach and using sp for such cases?

Reply:
Got it, but I'm not sure that answers my question question--it seems really easy to miss this detail in future development. How do we make sure that doesn't happen?

Reply:
I think we can mention this in  [General tips to make accessible](https://github.com/oppia/oppia-android/wiki/Accessibility-(A11y)-Guide#general-tips-to-make-app-accessible) guideline to use width in sp for recyclerview items instead of using wrap_content.

Reply:
@BenHenning Ideally we should never use `sp` for any non-text based dimension. But promoted story card is special case because here we have constraints like keeping all items width equal + the width should be scalable too and its doable only with `sp` correctly. 

So basically developers should use `dp` only and that too normally with `minWidth` / `minHeight` i.e.e avoid using fixed withd/height.

Reply:
I get your point @rt4914 but still I am thinking how we can add all of the content (3 text views inside a linear layout) will arrange with your proposed layout. As for our application we thumbnail images and we can't put text inside an image, like from UI perspective it won't looks good. Also, I still believe that with your approach of making a card of full width will re-create the issue #4684 as the thumbnail image sizes are not consistent.

Reply:
So I think we should at least make sure the accessibility guidelines call out this special case (i.e. the case of needing to use a fixed width container with text contained) along with an explanation for how to make it correct for accessibility.

Re: the banner approach, I think my main questions would be whether we can get the dragging functionality working there, and what the cycling experience is when using a screenreader. With regards to the contents, why wouldn't three text views & an image work? The first screenshot seems to do something quite similar.

Reply:
How can I update the accessibility guidelines in Wiki, or can you update it, if possible. Thanks

The exact guideline needs to be added as Follows:

- We can use width in SP if we have text inside a fixed width container. This will increase the size of container whenever we increase the font size, so the scaled text get enough size to fit inside a container.

Reply:
@BenHenning 
For banners the talkback will actually work with good experience. The main difference will be folloowing:

For sighted users
- The banners will be cyclic i.e. item-0,item1,item2,item0 and repeat

For talkback users
- the cyclic nature will stop
- the screenreader will start from item-0, item 1, item-2 and next it will go out of list.

Reply:
@BenHenning as you have suggested I have created a [google doc](https://docs.google.com/document/d/1FkZaFTT2hRo_3JShqhCwCZVVLGzUPiriW06Xme1EVaQ/edit?usp=sharing) to update the wiki. 

Also, I have filed a new issue #4703 to track the new approach suggested by Rajat

Reply:
Resolving this conversation as @rt4914 already replied to your comment and it's blocking PR from getting merged